File: src/xmlvm2js/System/EventHandler.js
/* Copyright (c) 2002-2011 by XMLVM.org
 *
 * Project Info:  http://www.xmlvm.org
 *
 * This program is free software; you can redistribute it and/or modify it
 * under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation; either version 2.1 of the License, or
 * (at your option) any later version.
 *
 * This library is distributed in the hope that it will be useful, but
 * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
 * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public
 * License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public
 * License along with this library; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301,
 * USA.
 */

qx.Class.define("System_EventHandler", {
  extend: System_Object,
  construct: function(targetObj, signatureIndex)
  {
      this.targetObj = targetObj;
      this.senderObj = 0;
      this.signatureIndex = signatureIndex;
  },
  statics:
  {
  	$__NEW___System_Object_int: function(targetObj, signatureIndex)
  	{
  		return new System_EventHandler(targetObj, signatureIndex);
  	}
  },
  members:
  {
    targetObj: 0,
    senderObj: 0,
    signatureIndex: 0
  }
});

File: src/wp7/Compatlib/System/EventHandler.java
/*
 * Copyright (c) 2004-2009 XMLVM --- An XML-based Programming Language
 * 
 * This program is free software; you can redistribute it and/or modify it under
 * the terms of the GNU General Public License as published by the Free Software
 * Foundation; either version 2 of the License, or (at your option) any later
 * version.
 * 
 * This program is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
 * details.
 * 
 * You should have received a copy of the GNU General Public License along with
 * this program; if not, write to the Free Software Foundation, Inc., 675 Mass
 * Ave, Cambridge, MA 02139, USA.
 * 
 * For more information, visit the XMLVM Home Page at http://www.xmlvm.org
 */

package Compatlib.System;

import org.xmlvm.XMLVMIgnore;
import org.xmlvm.XMLVMSkeletonOnly;

@XMLVMSkeletonOnly
public class EventHandler extends MulticastDelegate {

    @XMLVMIgnore
    public EventHandler(Object arg0, int arg1) {
        super(arg0, arg1);
    }

    public EventHandler(Object arg0, String arg1) {
        super();
    }

    public void Invoke(Object arg0, EventArgs arg1) {
        Object aobj[] = { arg0, arg1 };
        try {
            helpInvoke(aobj);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}
